### PR TITLE
webirc: Add KiwiIRC config

### PR DESCRIPTION
**Pull request by RyanSquared**
Prerequisite for https://github.com/hashbang/docker-webirc/pull/2.
Not exactly required but if we need to bump docker-webirc version it
would be nice to have this done first.

### CODE REVIEW

File: webirc/kustomization.yaml [hunk 1]
```diff
@@ -7,6 +7,12 @@ resources:
 replicas:
 - name: webirc
   count: 2
+configMapGenerator:
+- name: kiwiirc-config
+  options:
+    disableNameSuffixHash: true
+  files:
+  - kiwiirc-config.json
```
Comment (daurnimator) [inline webirc/kustomization.yaml:15]: could you put this in a `files/` subdirectory?

Should it actually be called `kiwiirc-config.json`? or just `config.json` ?

**Review by daurnimator — CHANGES_REQUESTED** [review]

File: webirc/kustomization.yaml [hunk 2]
```diff
@@ -7,6 +7,12 @@ resources:
 replicas:
 - name: webirc
   count: 2
+configMapGenerator:
+- name: kiwiirc-config
+  options:
+    disableNameSuffixHash: true
```
Comment (daurnimator) [inline webirc/kustomization.yaml:13]: why?

**Review by daurnimator — CHANGES_REQUESTED** [review]

File: webirc/resources.yaml [hunk 3]
```diff
@@ -28,6 +28,13 @@ spec:
           httpGet:
             path: /
             port: http
+        volumeMounts:
+        - name: kiwiirc-config
+          mountPath: /usr/share/nginx/html/static/
```
Comment (daurnimator) [inline webirc/resources.yaml:33]: Does this over-mount the whole directory?
Do we want to mount just a single file out of it?

Also consider passing `readOnly: true` (sic?)

Comment (RyanSquared) [reply webirc/resources.yaml:33]: I didn't think about that. I'm not sure how to do that.

Oddly enough, the diff states that the file mode would be `420`. That can't be right? Why would it do that?

```diff
===== apps/Deployment webirc/webirc ======
175a176,178
>         volumeMounts:
>         - mountPath: /usr/share/nginx/html/static/
>           name: kiwiirc-config
180a184,188
>       volumes:
>       - configMap:
>           defaultMode: 420
>           name: kiwiirc-config-h4m4gck68d
>         name: kiwiirc-config
===== /ConfigMap /kiwiirc-config-h4m4gck68d ======
0a1,44
> apiVersion: v1
> data:
>   config.json: |
>     {
>         "windowTitle": "hashbang web chat",
>         "useMonospace": true,
>         "startupScreen": "welcome",
>         "restricted": true,
>         "theme": "Dark",
>         "themes": [
>             { "name": "Default", "url": "static/themes/default" },
>             { "name": "Dark", "url": "static/themes/dark" },
>             { "name": "Coffee", "url": "static/themes/coffee" },
>             { "name": "GrayFox", "url": "static/themes/grayfox" },
>             { "name": "Nightswatch", "url": "static/themes/nightswatch" },
>             { "name": "Osprey", "url": "static/themes/osprey" },
>             { "name": "Radioactive", "url": "static/themes/radioactive" },
>             { "name": "Sky", "url": "static/themes/sky" },
>             { "name": "Elite", "url": "static/themes/elite" }
>         ],
>         "startupOptions": {
>             "greetingText": "hashbang web chat",
>             "websocket": "wss://irc.hashbang.sh/",
>             "channel": "#!",
>             "nick": "kiwi-n?"
>         },
>         "ircFramework": {
>             "auto_reconnect": true,
>             "auto_reconnect_wait": 4000,
>             "auto_reconnect_max_retries": 10
>         },
>         "buffers": {
>             "messageLayout": "compact",
>             "share_typing": false,
>             "show_emoticons": false,
>             "show_link_previews": false,
>             "inline_link_auto_previews": false
>         }
>     }
> kind: ConfigMap
> metadata:
>   labels:
>     app.kubernetes.io/instance: webirc
>   name: kiwiirc-config-h4m4gck68d
```

Comment (daurnimator) [reply webirc/resources.yaml]: why wouldn't it be 420?
and have you tried `readOnly: true` ?

Comment (RyanSquared) [reply webirc/resources.yaml]: I don't understand why the user would have permission to read it but not write it, and the group would have permission to write it and not read it.

I don't think we need to care about `readOnly: true` for a static webserver do we? Does it serve a practical purpose?

Comment (daurnimator) [reply webirc/resources.yaml]: 420 is decimal; equal to 644 in octal.

Comment (RyanSquared) [reply webirc/resources.yaml]: Alright.

But, I still don't see the reason to add readOnly given we don't have something that could write to it.

Comment (daurnimator) [reply webirc/resources.yaml]: That's precisely the point though: unless you want something writing to it; set readOnly to true.

Comment (RyanSquared) [reply webirc/resources.yaml]: I don't see any harm in not doing it.

**Comment by RyanSquared** [comment]
If we're going to set `readOnly` for configmaps we should do it across all configmaps in the repository, and probably start up a style guide for that kind of thing. That can come in a later commit. Is there anything blocking this from merging?

**Review by daurnimator — APPROVED** [review]

**Pull request merged**